File: B13/Subhendu/Array/noisy-palindrome.py
class Solution:
    def solve(self, s):
        ### https://binarysearch.com/problems/Noisy-Palindrome
        l = 0
        r = len(s) - 1
        chars = "abcdefghijklmnopqrstuvwxyz"
        while l < r:
            while l < r and s[l] not in chars:
                l += 1

            while l < r and s[r] not in chars:
                r -= 1

            if s[l] != s[r]:
                return False
            l += 1
            r -= 1
        return True

File: B13/Hemanth/Prefix_and_Suffix/Noisy_Palindrome.java
//https://binarysearch.com/problems/Noisy-Palindrome


import java.util.*;

class Solution {
    public boolean solve(String s) {
        String ans="";
        for(char ch:s.toCharArray()){
            if(Character.isLowerCase(ch)){
                ans+=ch;
            }
        }
        return palin(ans);
    }

    public boolean palin(String str){
        for(int i=0;i<str.length();i++){
            char a = str.charAt(i);
            char b = str.charAt(str.length()-i-1);
            if(a!=b){
                return false;
            }
        }
        return true;
    }
}

